### PR TITLE
Add in Attestation Validity Check

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Bug Fix

**What does this PR do? Why is it needed?**

- [x] Validate Attestation Indices in Batch Attestation Processing.
- [x] Add Regression test to validate indices from attestation. 

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

File: beacon-chain/core/blocks/attestation.go [hunk 1]
```diff
@@ -306,6 +252,60 @@ func VerifyAttestations(ctx context.Context, beaconState *stateTrie.BeaconState,
 	return verifyAttestationsWithDomain(ctx, beaconState, postForkAtts, currDomain)
 }
 
+// VerifyAttestation converts and attestation into an indexed attestation and verifies
+// the signature in that attestation.
+func VerifyAttestation(ctx context.Context, beaconState *stateTrie.BeaconState, att *ethpb.Attestation) error {
+	if att == nil || att.Data == nil {
```
Comment (shayzluf) [inline beacon-chain/core/blocks/attestation.go:258]: can we check aggregation bits is not 0 here ?
wasteful to convert to indexed when this is the case

File: beacon-chain/core/blocks/attestation.go [hunk 2]
```diff
@@ -306,6 +252,60 @@ func VerifyAttestations(ctx context.Context, beaconState *stateTrie.BeaconState,
 	return verifyAttestationsWithDomain(ctx, beaconState, postForkAtts, currDomain)
 }
 
+// VerifyAttestation converts and attestation into an indexed attestation and verifies
+// the signature in that attestation.
+func VerifyAttestation(ctx context.Context, beaconState *stateTrie.BeaconState, att *ethpb.Attestation) error {
+	if att == nil || att.Data == nil || att.AggregationBits.Count() == 0 {
+		return fmt.Errorf("nil or missing attestation data: %v", att)
```
Comment (shayzluf) [inline beacon-chain/core/blocks/attestation.go:259]:
```suggestion
		return fmt.Errorf("nil, missing attestation data or no attesting indices: %v", att)
```

**Comment by nisdas** [comment]
No , there was a valid indexed attestation check that was missing completely. It deviated from the spec and our old core methods. When changing it to batch verify attestations we missed this whole check.

**Comment by terencechain** [comment]
> No , there was a valid indexed attestation check that was missing completely. It deviated from the spec and our old core methods. When changing it to batch verify attestations we missed this whole check.

I see the `VerifyAttestation

**Pull request merged**